### ISSUE TITLE
Pre-start command permissioning is inconsistent

### ISSUE THREAD

**Issue by hashbrowncipher** [EvanKrall/puppet-initscript#14]
In upstart and systemd styles, the before_command commands run under the accounts specified in `setuid` and `User=`, respectively. In the sysv styles, before_command commands run as root.

I actually think there's greater utility to running pre-start commands as root. Take, for instance, a pre-start script which does `mkdir /run/myprogram; chown myuser /run/myprogram`. This should typically be run as root.

**Comment by EvanKrall** [comment]
I agree that it's useful to be able to run pre-start commands as root. Would be nice to have it be an option, though it would add complexity.

[This superuser post](http://superuser.com/questions/722215/upstart-run-service-as-unprivileged-user-and-pre-start-script-as-root) has some good ideas:
- don't use setuid, instead use sudo or `setuidgid` or similar in the exec/script block.
- create another service with `start on starting realservice`, with no setuid stanza.
